.circleci/README.md

```diff
@@ -0,0 +1,69 @@
+in your repository. Each is described below, along with a description of environment
+variables that you can define in the CircleCI project settings to customize the build.
+
+## Where do I define the variables?
+For each set of variables, some you can define in the config.yml, in the `environment`
+section for a workflow step, for example, here is setting the variable `TIMEZONE`
+to be defined as `/usr/share/zoneinfo/America/Los_Angeles` available during the
+"setup" step:
+
+```bash
+  setup:
+    environment:
+      - TIMEZONE: "/usr/share/zoneinfo/America/Los_Angeles"
+```
+
+or you can define variables in the [CircleCI project settings](https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-project). 
+The latter is recommended for sensitive information. 
+
+For all configuration descriptions below, **all variables** from both tables 
+can be set in just the CircleCI project settings, if desired.
+
+# Templates
+
+## config.yml
+
+This is a base, or default configuration to build.
+
+**Variables that must be set in CircleCI Project Settings**
+
+|Variable|Description|Required|Default|
+|---|---|---|---|
+|`DOCKER_USER`|Docker username to deploy to Docker Hub. The build will not deploy without it.|no|unset|
+|`DOCKER_PASS`|Docker password to deploy to Docker Hub. The build will not deploy without it.|no|unset|
+
+**Variables that can be set in config.yml**
+ 
+|Variable|Description|Required|Default|
+|---|---|---|---|
+|`CONTAINER_NAME`|The name of the Docker Hub container to deploy|no|`<ORG>/<REPO>`|
+|`DOCKER_TAG`|A custom tag for the deployed image.|no|the first 10 characters of the commit.|
+
+
+## config.singularity
+
+This configuration is equivalent to the default config.yml, with the additional change in permissions of the
+repo2docker folder to allow for the user to write given export into a Singularity container. If you want your
+container notebooks to be usable on a shared resource, you should choose this configuration. Note that the
+save happens in a temporary location provided by the container, and so you would want to save your notebooks as an export to your local machine, or to another location on your shared resource.
+
+**Variables that must be set in CircleCI Project Settings**
+
+|Variable|Description|Required|Default|
+|---|---|---|---|
+|`DOCKER_USER`|Docker username to deploy to Docker Hub. The build will not deploy without it.|no|unset|
+|`DOCKER_PASS`|Docker password to deploy to Docker Hub. The build will not deploy without it.|no|unset|
+
+**Variables that can be set in config.yml**
+
+|Variable|Description|Required|Default|
+|---|---|---|---|
+|`CONTAINER_NAME`|The name of the Docker Hub container to deploy|no|`<ORG>/<REPO>`|
+|`DOCKER_TAG`|A custom tag for the deployed image.|no|the first 10 characters of the commit.|
+|`NOTEBOOK_PERMISSION`|The permission to (recursively) give the notebook, to allow write.|no|777|
```

Comment:
Should we remove NOTEBOOK_ from the permission and username variable? Neither really have anything to do with the fact that we use notebooks. As in we'd still have to set them if we were using RStudio or OpenRefine.

Maybe | `PERMISSION` | The permissions to (recursively) set on the repository contents copied into the container |?

Reply:
Sure thing!

Reply:
However the NOTEBOOK is relevant - these variables specifically change just the folder for the notebook housing and permission for it. The `NOTEBOOK` also serves as a namespace, so there is less liklihood of conflict with some other user specific variable. If you want to think of a different prefix to replace this, I'm happy to change it. I think notebook is more appropriate than something like container, because it is truly just a username for the notebook, and permission for the notebook folder. That may be different outside of using repo2docker, but right now that's all this recipe does, and so the variable names are clear.

Reply:
Maybe `REPO2DOCKER_` as prefix? There doesn't have to be a notebook in the repository and this would set the permissions on all things that are copied over right?

Reply:
yes I like that too. 

Yes that's correct. We are changing the permission of the user "joyvan's" folder in $HOME:

```bash
docker exec repo2docker chmod -R ${REPO2DOCKER_PERMISSION} "/home/${REPO2DOCKER_USERNAME}"
```
and we have to do this for Singularity, otherwise the notebooks aren't usable for the general user.